Sources/Testing/Running/Configuration.swift

```diff
@@ -137,38 +137,25 @@ public struct Configuration: Sendable {
 
   // MARK: - Test selection
 
-  /// The selected tests to run, if any.
+  /// A function that handles filtering tests.
   ///
-  /// This property should be used for testing membership (whether a test ID has
-  /// been selected) since it is more optimized for that use case. It also
-  /// provides the backing storage for ``selectedTestIDs``.
-  ///
-  /// This property is optional and defaults to `nil` because it is possible to
-  /// select specific tests to run but not provide any tests in that list. That
-  /// is a supported use case: it results in zero tests being run and no issues
-  /// recorded.
-  ///
-  /// A practical example of when this situation can happen is when testing is
-  /// configured via an Xcode Test Plan, the "Automatically Include New Tests"
-  /// option is disabled, and zero tests are enabled.
-  var selectedTests: Test.ID.Selection?
-
-  /// The IDs of the selected tests to run, if any.
-  ///
-  /// This property is optional and defaults to `nil` because it is possible to
-  /// select specific tests to run but not provide any tests in that list. That
-  /// is a supported use case: it results in zero tests being run and no issues
-  /// recorded.
-  ///
-  /// A practical example of when this situation can happen is when testing is
-  /// configured via an Xcode Test Plan, the "Automatically Include New Tests"
-  /// option is disabled, and zero tests are enabled.
-  public var selectedTestIDs: Set<Test.ID>? {
-    get {
-      selectedTests?.testIDs
-    }
-    set {
-      selectedTests = newValue.map { .init(testIDs: $0) }
-    }
+  /// - Parameters:
+  ///   - test: An test that needs to be filtered.
+  /// - Returns: A Boolean value representing if the test satisfied the filter.
+  public typealias TestFilter = @Sendable (Test) -> Bool
+
+  /// The test filter to which tests should be filtered when run.
+  public var testFilter: TestFilter?
+
+  /// The granularity to enforce test filtering.
+  /// 
+  /// By default, all tests are run and no filter is set.
+  @_spi(ExperimentalTestFilter)
```

Comment:
We can skip the `@_spi` marker here because `Configuration` already has one and I think that's adequate